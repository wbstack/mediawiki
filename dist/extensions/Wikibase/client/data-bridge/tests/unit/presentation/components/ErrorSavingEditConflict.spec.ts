import MessageKeys from '@/definitions/MessageKeys';
import ErrorSavingEditConflict from '@/presentation/components/ErrorSavingEditConflict.vue';
import EventEmittingButton from '@/presentation/components/EventEmittingButton.vue';
import { shallowMount } from '@vue/test-utils';

describe( 'ErrorSavingEditConflict', () => {
	const $messages = {
		KEYS: MessageKeys,
		getText: jest.fn( ( key: string ) => `⧼${key}⧽` ),
	};

	it( 'matches the snapshot', () => {
		const wrapper = shallowMount( ErrorSavingEditConflict, { mocks: { $messages } } );

		expect( wrapper.element ).toMatchSnapshot();
	} );

	it( 'maps reload button click to reload event', async () => {
		const wrapper = shallowMount( ErrorSavingEditConflict );

		wrapper.findComponent( EventEmittingButton ).vm.$emit( 'click' );
		expect( wrapper.emitted( 'reload' ) ).toHaveLength( 1 );
	} );
} );
