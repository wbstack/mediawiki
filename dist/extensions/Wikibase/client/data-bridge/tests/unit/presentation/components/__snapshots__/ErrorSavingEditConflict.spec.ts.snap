// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSavingEditConflict matches the snapshot 1`] = `
<div
  class="wb-db-error-saving-edit-conflict"
>
  <h2
    class="wb-db-error-saving-edit-conflict__heading"
  >
    ⧼wikibase-client-data-bridge-edit-conflict-error-heading⧽
  </h2>
  <icon-message-box-stub
    class="wb-db-error-saving-edit-conflict__message"
    inline="false"
    type="error"
  />
  <event-emitting-button-stub
    class="wb-db-error-saving-edit-conflict__reload"
    disabled="false"
    message="⧼wikibase-client-data-bridge-error-reload-page⧽"
    newtab="false"
    preventdefault="true"
    size="M"
    squary="false"
    type="primaryProgressive"
  />
</div>
`;
